### PR TITLE
WIP Apollo-Server migration

### PR DESCRIPTION
**Pull request by emilioward**
This PR adds in the Apollo-Server configuration, it's currently a work in progress migrating the api from graphql-express to apollo-server.

To use the Apollo-Server launch the application, you can switch to the graphiql version by simply modifying the URL to:

`http://localhost:4000/graphiql?query=%0A%0A%7B%0A%20%20hello%7B%0A%20%20%20%20text%0A%20%20%7D%0A%7D%0A%0A` 

Or swap to the apollo-server by removing the "i" in the URL:

`http://localhost:4000/graphql?query=%0A%0A%7B%0A%20%20hello%7B%0A%20%20%20%20text%0A%20%20%7D%0A%7D%0A%0A`.

### CODE REVIEW

**Comment by emilioward** [comment]
[Move to Apollo graphql-server backend (2)](https://trello.com/c/w1lSgEl4/91-move-to-apollo-graphql-server-backend-2)

**Comment by samiwel** [comment]
Looks like I was just reading this wrong yesterday. See http://dev.apollodata.com/tools/graphql-server/graphiql.html#graphiqlOptions. Basically there needs to be another endpoint e.g. `/graphiql`
and the docs say that the `endpointURL` option should point to the __GraphQL__ endpoint which GraphiQL serves! so that should be the `/graphql`.

Then just do a check for the `process.env.NODE_ENV === "development"` before adding the `/graphiql` endpoint!

Hope that makes sense! The names are just too similar!

File: src/server.js [hunk 1]
```diff
@@ -18,6 +21,12 @@ app.use(GRAPHIQL_ENDPOINT,
   graphiql: GRAPHIQL_ENABLED,
 }));
 
+app.use('/graphql', cors(), graphqlExpress({ schema: schema }))
+
+app.use('/mode', (req, res) => {
```
Comment (samiwel) [inline src/server.js]: Get rid of mode

File: src/server.js [hunk 2]
```diff
@@ -18,6 +21,12 @@ app.use(GRAPHIQL_ENDPOINT,
   graphiql: GRAPHIQL_ENABLED,
 }));
 
+app.use('/graphql', cors(), graphqlExpress({ schema: schema }))
```
Comment (samiwel) [inline src/server.js]:
```
if (process.env.NODE_ENV === 'development') {
app.use('/graphiql', cors(), graphiqlExpress({endpointURL: '/graphql' }))
}
```

File: src/createGraphQLServer.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+var graphqlHTTP = require("express-graphql");
```
Comment (samiwel) [inline src/createGraphQLServer.js]: Get rid of this file altogether.

File: package.json [hunk 4]
```diff
@@ -8,18 +8,19 @@
   },
   "dependencies": {
     "colors": "^1.1.2",
+    "cors": "^2.8.3",
     "dotenv": "^4.0.0",
     "express": "^4.15.3",
     "express-graphql": "^0.6.5",
```
Comment (samiwel) [inline package.json]: yarn remove

File: src/createGraphQLServer.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+const graphqlExpress = require('graphql-server-express').graphqlExpress;
+var schema = require("./schema");
```
Comment (hamishtaplin) [inline src/createGraphQLServer.js]: use `const` for consistency

File: src/createGraphQLServer.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+var {
```
Comment (hamishtaplin) [inline src/createGraphQLServer.js]: `const`

File: src/createGraphQLServer.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  GRAPHIQL_ENABLED,
+  GRAPHIQL_ENDPOINT,
+  GRAPHIQL_PRETTY } = require('./settings').graphiql;
+
+module.exports = () => {
+  const ENV = process.env.NODE_ENV || 'production';
+
+  let result;
+  if (ENV === 'production') {
+    result = graphqlExpress({ schema: schema });
```
Comment (hamishtaplin) [inline src/createGraphQLServer.js]: Can you use [object shorthand](http://www.benmvp.com/learning-es6-enhanced-object-literals/) here?

File: src/createGraphQLServer.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  } else {
+    result = graphqlHTTP({
+    schema: schema,
```
Comment (hamishtaplin) [inline src/createGraphQLServer.js]: Object shorthand again

File: src/server.js [hunk 9]
```diff
@@ -1,5 +1,8 @@
+const ENV = process.env.NODE_ENV || 'production'
+
 var express = require("express");
```
Comment (hamishtaplin) [inline src/server.js]: Use `const` consistently.

File: src/server.js [hunk 10]
```diff
@@ -1,22 +1,23 @@
-var express = require("express");
-var graphqlHTTP = require("express-graphql");
-var schema = require("./schema");
-var colors = require("colors");
+const ENV = process.env.NODE_ENV || 'production'
+
+const express = require("express");
+const graphqlExpress = require('graphql-server-express').graphqlExpress;
+const graphiqlExpress = require('graphql-server-express').graphiqlExpress;
```
Comment (WickyNilliams) [inline src/server.js]: Can simplify this to:

```js
const { graphqlExpress, graphiqlExpress } = require("graphql-server-express");
```

File: src/server.js [hunk 11]
```diff
@@ -1,22 +1,23 @@
-var express = require("express");
-var graphqlHTTP = require("express-graphql");
-var schema = require("./schema");
-var colors = require("colors");
+const ENV = process.env.NODE_ENV || 'production'
```
Comment (WickyNilliams) [inline src/server.js]: This doesn't seem to be use anywhere. Can be removed

**Pull request closed without merging**